tests/helpers/test_config_validation.py

```diff
@@ -1332,3 +1332,26 @@ def test_currency():
 
     for value in ("EUR", "USD"):
         assert schema(value)
+
+
+def test_time_period_str():
+    """Test time_period_str validator and transformer."""
+    schema = vol.Schema(cv.time_period_str)
+
+    for value in [
+        None,
+        datetime.now().time(),
+        "2016-11-23",
+        "2016-11-23T18:59:08",
+        " 1 23:42:25.987654",
+        "1 1 23:42:25.987654",
+    ]:
+        with pytest.raises(vol.MultipleInvalid):
+            schema(value)
```

Comment:
I think it makes more sense to split these into separate tests and parameterize these.

Reply:
I don't understand. Can you elaborate?

Reply:
Make a separate test for invalid tests vs valid tests below (instead of putting them into a single test). Once you have split that, you can use the parameterize capability of PyTest to run multiple cases (instead of using a for loop).